agent.go

```diff
@@ -98,17 +103,22 @@ type sandboxStorage struct {
 
 type sandbox struct {
 	sync.RWMutex
+	ctx context.Context
+
+	id          string
+	hostname    string
+	containers  map[string]*container
+	channel     channel
+	network     network
+	wg          sync.WaitGroup
+	sharedPidNs namespace
+	mounts      []string
+	subreaper   reaper
+	server      *grpc.Server
+
+	// Set when server needs to be shut down
+	shutdown chan bool
```

Comment:
Instead you could use:
```go
shutdown chan struct{}
```
since you don't care about the boolean value anywhere in the code.